File: OnVac.ObjectModel/Entities/ArchiviazioneDIRV/Intestazione.vb
Imports System.Collections.Generic

Namespace Entities.ArchiviazioneDIRV

    <System.Xml.Serialization.XmlTypeAttribute()>
    <System.Xml.Serialization.XmlRootAttribute("Intestazione")>
    <System.SerializableAttribute()>
    Public Class Intestazione

        Public IdMessaggio As String
        Public TmstMessaggio As String
        Public Mittente As String
        Public Servizio As String
        Public Destinatario As String
        Public UserId As String
        Public Password As String
        Public RichiestaApplicativa As RichiestaApplicativa
        Public RispostaApplicativa As RispostaApplicativa

        Public Sub New()
            Me.RichiestaApplicativa = New RichiestaApplicativa()
            Me.RispostaApplicativa = New RispostaApplicativa()
        End Sub

    End Class

    Public Class RichiestaApplicativa

        Public Allegati As List(Of Allegato)

        Public Sub New()
            Me.Allegati = New List(Of Allegato)()
        End Sub

    End Class

    Public Class RispostaApplicativa

        Public Allegati As List(Of Allegato)
        Public Messaggio As String

        Public Sub New()
            Me.Allegati = New List(Of Allegato)()
        End Sub

    End Class

    Public Class Allegato

        Public Nome As String

        Public Sub New()
        End Sub

        Public Sub New(nome As String)
            Me.Nome = nome
        End Sub

    End Class

End Namespace

File: OnVac.DIRV/ArchiviazioneSostitutiva/Headers/Intestazione.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Onit.OnAssistnet.OnVac.DIRV.ArchiviazioneSostitutiva.Headers
{
    /// <summary>
    /// La classe Intestazione rappresenta l'XML_MESSAGE da inviare al servizio
    /// </summary>
    [Serializable]
    public class Intestazione
    {
        /// <summary>
        /// Identificativo univoco del messaggio. 
        /// Non obbligatorio, se non impostato lo valorizza il servizio di archiviazione
        /// </summary>
        public string IdMessaggio { get; set; }
        
        /// <summary>
        /// Timestamp del messaggio. 
        /// Non obbligatorio, se non impostato lo valorizza il servizio di archiviazione
        /// </summary>
        public string TmstMessaggio { get; set; }
        
        /// <summary>
        /// Richiedente del servizio. Obbligatorio.
        /// </summary>
        public string Mittente { get; set; }
        
        /// <summary>
        /// Nome del servizio. Non è obbligatorio per i servizi specifici (come il servizio di archiviazione)
        /// </summary>
        public string Servizio { get; set; }
        
        /// <summary>
        /// Destinatario del servizio. Non obbligatorio
        /// </summary>
        public string Destinatario { get; set; }
        
        /// <summary>
        /// Id del richiedente, per i servizi che richiedono autenticazione tramite password.
        /// </summary>
        public string UserId { get; set; }
        
        /// <summary>
        /// Password del richiedente, per i servizi che richiedono autenticazione tramite password.
        /// </summary>
        public string Password { get; set; }

        /// <summary>
        /// Oggetto contenente allegati di richiesta.
        /// </summary>
        public RichiestaApplicativa RichiestaApplicativa { get; set; }
        
        /// <summary>
        /// Oggetto contenente allegati di risposta.
        /// </summary>
        public RispostaApplicativa RispostaApplicativa { get; set; }

        public Intestazione()
        {
            this.RichiestaApplicativa = new RichiestaApplicativa();
        }
    }
}
